pkg/wharfapi/client.go

```diff
@@ -95,73 +96,68 @@ func (c *Client) getUnmarshal(path string, q url.Values, response interface{}) e
 	if err != nil {
 		return err
 	}
-	err = json.NewDecoder(ioBody).Decode(response)
-	if err != nil {
-		return err
-	}
-	return ioBody.Close()
+	defer ioBody.Close()
```

Comment:
I made a test impl in 1a3ab90

I'm not that big of a fan of the implementation, and getting more keen in just ignoring these close errors as we don't really care if they fail. Conventionally speaking, I've not seen any codebase handle close errors, so it's not easy to find inspiration.

What do you think of the 1a3ab90 implementation?

Reply:
It looks better than what I was able to think off 👍🏻
Maybe it doesn't require a named return value to be passed in if you're doing it through a pointer?
Could make it a bit nicer.

> just ignoring these close errors as we don't really care if they fail. Conventionally speaking, I've not seen any codebase handle close errors, so it's not easy to find inspiration.

Yeah, this is why I put it as a nit as well. If it gets too messy it's not really worth it. I think `1a3ab90` is nice enough even with named return values though, so I'd vote for keeping it 👍🏻

Reply:
The named returns was to just save a line in the file. It hurts readability as some of them are really far to the end of a long line. But question is if the newly made convention of naming it `finalErr` is having the naked return is enough visual clues?

```go

func decodeJSONAndClose(r io.ReadCloser, obj interface{}) (finalErr error) {
	defer closeAndSetError(r, &finalErr)
	finalErr = json.NewDecoder(r).Decode(obj)
	return
}

// vs:

func decodeJSONAndClose(r io.ReadCloser, obj interface{}) (error) {
	var finalErr error
	defer closeAndSetError(r, &finalErr)
	finalErr = json.NewDecoder(r).Decode(obj)
	return finalErr
}
```

Reply:
I think it's a short enough function that it doesn't require much extra or any at all to understand what it's doing.

I was mainly thinking of when calling the function. Not sure how if it would affect the way it works, haven't tried it out myself.

If either works then I don't have any strong preference for either. Named feels like it's easier to tell from the declaration, but non-named is more regular and has slightly less bloat. Fully leaving it up to you 👍🏻

Reply:
The latter case doesn't even work, it turns out. I've added tests to verify this.

It's like it's doing the following:

```go
func decodeJSONAndClose(r io.ReadCloser, obj interface{}) (_resultParam0 error) {
	var finalErr error
	defer closeAndSetError(r, &finalErr)
	finalErr = json.NewDecoder(r).Decode(obj)
	_resultParam0 = finalErr
	return
}
```

Reply:
Strange stuff, nice with tests to verify 👍🏻